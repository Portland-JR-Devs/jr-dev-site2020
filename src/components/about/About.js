import React from 'react';
import './about.scss'

const About = (props) => {
    return(
        <>
            <div className='about-container'>
            <h1>ABOUT US</h1>
            <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam eget sem congue, tristique nulla in, mollis lorem. Vivamus est nulla, dapibus a nulla et, iaculis iaculis augue. Donec pretium risus eget turpis vestibulum, et malesuada leo aliquam. Interdum et malesuada fames ac ante ipsum primis in faucibus. Nullam pretium dignissim gravida. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Sed risus turpis, gravida quis libero vitae, dictum rutrum velit. Ut neque turpis, gravida a magna vitae, sagittis posuere enim. Integer viverra ante eros, egestas feugiat nibh suscipit eu.</p>

            <p>Morbi facilisis sapien eget purus hendrerit condimentum. Fusce ultricies quis lectus ut tempus. Praesent condimentum, elit ac ullamcorper finibus, urna turpis pharetra ante, non tincidunt nulla diam sed magna. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Nam fermentum convallis velit vel ullamcorper. Nunc et augue ante. Nulla eget nulla sollicitudin, fringilla tellus sit amet, porttitor quam. Mauris blandit congue tortor, sed venenatis lacus tincidunt quis. Proin et ipsum tortor. Quisque tempor malesuada ultrices. Sed sem magna, dapibus at nisi ullamcorper, tincidunt aliquam turpis. Suspendisse nisl sapien, convallis id turpis in, accumsan dignissim turpis. Nulla interdum ut justo quis ultrices.</p>
        </div>
        </>
    )
}

export default About;