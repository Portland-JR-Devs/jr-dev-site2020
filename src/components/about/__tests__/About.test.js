import React from "react"
import renderer from "react-test-renderer"
// import react-testing methods
import { render, screen, fireEvent } from "@testing-library/react"

// add custom jest matchers from jest-dom
import "@testing-library/jest-dom/extend-expect"

import AboutComponent from "../About"

import About from "../About"

describe("About", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<About />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  test("shows component title", () => {
    // ARRANGE
    render(<AboutComponent />)

    const regexForTitle = /About Us/i

    // Get the element that has the text we expect.
    const elementWithExpectedText = screen.queryByText(regexForTitle)

    // ACT
    // No user action involved here so no action to take.

    // ASSERT
    // Confirm the element has the expected text
    expect(elementWithExpectedText).toBeInTheDocument()
  })
})

// test("shows component title", () => {
//   // ARRANGE
//   render(<AboutComponent />)

//   const regexForTitle = /About Us/i

//   // Get the element that has the text we expect.
//   const elementWithExpectedText = screen.queryByText(regexForTitle)

//   // ACT
//   // No user action involved here so no action to take.

//   // ASSERT
//   // Confirm the element has the expected text
//   expect(elementWithExpectedText).toBeInTheDocument()
// })
