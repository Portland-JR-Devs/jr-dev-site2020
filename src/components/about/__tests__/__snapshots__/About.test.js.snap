// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About renders correctly: About.test.js 1`] = `
<div
  className="about-container"
>
  <h2
    id="about-us-header"
  >
    About Us
  </h2>
  <p>
    For entry level developers, people interested in starting a new career in tech, and anyone wanting to encourage and help newer developers. Join us to hear relevant speakers, have a slice of pizza, and make new connections! We look forward to meeting you. If youâ€™d like to be a part of our Slack channel, you can be a part of the 
    <a
      href="http://bit.ly/JrDevSlackInvite"
    >
      conversation here
    </a>
    .
  </p>
  <p>
    PDX Jr. Devs is a community of developers comprised of people that are getting their start in tech.
  </p>
</div>
`;
